src/main/scala/mesosphere/marathon/health/HealthCheck.scala

```diff
@@ -122,4 +122,36 @@ object HealthCheck {
   val DefaultMaxConsecutiveFailures = 3
   val DefaultIgnoreHttp1xx = false
   val DefaultPort = None
+
+  implicit val healthCheck = validator[HealthCheck] { hc =>
+    (hc.portIndex.nonEmpty is true) or (hc.port.nonEmpty is true)
+    hc is validProtocol
+  }
+
+  /**
```

Comment:
This comment here is:
- outdated: since the problem with the seq is not valid any longer
- in the wrong place: it probably should be moved to V2AppUpdate.appDefinitionValidator, where we _don't_ do `appDef.healthChecks is valid`